python/dgl/graphbolt/impl/torch_based_feature_store.py

```diff
@@ -234,6 +235,117 @@ def __repr__(self) -> str:
         )
 
 
+class DiskBasedFeature(Feature):
+    r"""A wrapper of disk based feature.
+
+    Initialize a disk based feature fetcher by a numpy file.
+
+    Parameters
+    ----------
+    path : string
+        The path to the numpy feature file.
+        Note that the dimension of the numpy should be greater than 1.
+    Examples
+    --------
+    >>> import torch
+    >>> from dgl import graphbolt as gb
+    >>> torch_feat = torch.arange(10).reshape(2, -1)
+    >>> feature = gb.DiskBasedFeature(torch_feat)
+    >>> feature.read(torch.tensor([0]))
+    tensor([[0, 1, 2, 3, 4]])
+    >>> feature.size()
+    torch.Size([5])
+    """
+
+    def __init__(self, path: str, metadata: Dict = None):
+        super().__init__()
+        mmap_mode = "r+"
+        self._tensor = torch.from_numpy(
```

Comment:
can this be wrapped into existing `read_data()` for wider usage such as `OnDiskDataset`?
https://github.com/dmlc/dgl/blob/8e3d810127e01dff388cea588e917c3f1c9943a8/python/dgl/graphbolt/internal/utils.py#L25